src/test/java/com/yelp/nrtsearch/server/grpc/IndexNamesAddDocumentsTest.java

```diff
@@ -0,0 +1,182 @@
+
+import com.yelp.nrtsearch.server.ServerTestCase;
+import io.grpc.testing.GrpcCleanupRule;
+import java.io.IOException;
+import java.io.Reader;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+/**
+ * This tests adding documents to multiple indices at once by specifying a list of index names
+ * rather than setting the index name individually for each document.
+ */
+public class IndexNamesAddDocumentsTest extends ServerTestCase {
+  private static final String INDEX_1 = "test_index_1";
+  private static final String INDEX_2 = "test_index_2";
+  private static final String INDEX_3 = "test_index_3";
+
+  private static final Map<String, String> INDEX_TO_FIELD_DEF =
+      Map.of(
+          INDEX_1, "/registerFieldsMultiIndexIndexing1.json",
+          INDEX_2, "/registerFieldsMultiIndexIndexing2.json");
+
+  private static final Map<String, String> INDEX_TO_DOCS =
+      Map.of(INDEX_1, "/addDocsMultiIndexIndexing1.csv");
+
+  @ClassRule public static final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+  @Override
+  public List<String> getIndices() {
+    return Arrays.asList(INDEX_1, INDEX_2);
+  }
+
+  @Override
+  public FieldDefRequest getIndexDef(String name) throws IOException {
+    String resourceFile = INDEX_TO_FIELD_DEF.getOrDefault(name, "");
+    if (resourceFile.isEmpty()) {
+      throw new IllegalStateException("Unexpected index: " + name);
+    }
+    return getFieldsFromResourceFile(resourceFile);
+  }
+
+  @Override
+  protected void initIndices() throws Exception {
+    for (String indexName : getIndices()) {
+      LuceneServerGrpc.LuceneServerBlockingStub blockingStub = getGrpcServer().getBlockingStub();
+
+      // create the index
+      blockingStub.createIndex(CreateIndexRequest.newBuilder().setIndexName(indexName).build());
+
+      // Register fields with the same field definition for both indices to ensure compatibility
+      FieldDefRequest fieldDefRequest = getIndexDef(INDEX_1);
+      blockingStub.registerFields(
+          FieldDefRequest.newBuilder(fieldDefRequest).setIndexName(indexName).build());
+
+      // Apply settings and start the index
+      blockingStub.liveSettings(getLiveSettings(indexName));
+      blockingStub.startIndex(StartIndexRequest.newBuilder().setIndexName(indexName).build());
+    }
+  }
+
+  /**
+   * Tests adding documents to multiple indices using indexNames list rather than individual
+   * indexName.
+   */
+  @Test
+  public void testMultiListIndexAddDocuments() throws Exception {
+    // Create documents using the list of target indices
+    List<String> targetIndices = Arrays.asList(INDEX_1, INDEX_2);
```

Comment:
Nit: use `List.of()` instead of `Arrays.asList()` since it's a newer syntax and slightly shorter.